.github/workflows/reusable-smoke-test.yml

```diff
@@ -55,19 +55,31 @@ jobs:
           distribution: temurin
           java-version: 17
 
-      - name: Test
+      - name: Build
         env:
           GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
           GE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
           GE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: ":smoke-tests:test -PsmokeTestSuite=${{ matrix.smoke-test-suite }}${{ inputs.no-build-cache && ' --no-build-cache' || '' }}"
+          # running suite "none" compiles everything needed by smoke tests without executing any tests
+          arguments: ":smoke-tests:test -PsmokeTestSuite=none --no-daemon ${{ inputs.no-build-cache && ' --no-build-cache' || '' }}"
           # only push cache for one matrix option per OS since github action cache space is limited
           cache-read-only: ${{ inputs.cache-read-only || matrix.smoke-test-suite != 'tomcat' }}
           # gradle enterprise is used for the build cache
           gradle-home-cache-excludes: caches/build-cache-1
 
+      - name: Test
+        env:
+          GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
+          GE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+          GE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: ":smoke-tests:test -PsmokeTestSuite=${{ matrix.smoke-test-suite }}${{ inputs.no-build-cache && ' --no-build-cache' || '' }}"
+          # Build step takes care of caching
+          cache-disabled: true
```

Comment:
I don't know whether this is needed. Only the first use of `gradle-build-action` restores caches, idk about saving them. I guess it won't hurt to disable caching here.

Reply:
to avoid this confusion, I've been adding a single `Set up Gradle cache` step in cases where we execute two or more gradle commands, e.g.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/85061691fe1a0633be15f37ac461d882bbec3757/.github/workflows/reusable-examples.yml#L25-L37